### PR TITLE
fix(module:tooltip,popover,popconfirm): fix hydration error

### PR DESCRIPTION
**Pull request by HyperLife1119**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #8510, #8181, #8406

https://angular.dev/errors/NG0500

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/tooltip/base.ts [hunk 1]
```diff
@@ -116,13 +112,18 @@ export abstract class NzTooltipBaseDirective implements OnChanges, OnDestroy, Af
...
+  protected hostView = inject(ViewContainerRef);
+  protected renderer = inject(Renderer2);
+  protected noAnimation = inject(NzNoAnimationDirective, { host: true, optional: true });
+  protected nzConfigService = inject(NzConfigService);
+
+  constructor(protected componentType: Type<NzTooltipBaseComponent>) {
+    afterNextRender(() => {
+      this.createComponent();
+      this.registerTriggers();
+    });
```
Comment (HyperLife1119) [inline components/tooltip/base.ts:125]: Key: afterNextRender will only be executed in the client, which ensures that components are only dynamically created in the browser.

File: components/tooltip/base.ts [hunk 2]
```diff
@@ -171,7 +167,8 @@ export abstract class NzTooltipBaseDirective implements OnChanges, OnDestroy, Af
    * Create a dynamic tooltip component. This method can be override.
    */
   protected createComponent(): void {
-    const componentRef = this.componentRef;
+    const componentRef = this.hostView.createComponent(this.componentType);
```
Comment (HyperLife1119) [inline components/tooltip/base.ts:170]: Key: non-eager dynamic creation of components.

File: components/tooltip/base.ts [hunk 3]
```diff
@@ -116,13 +113,24 @@ export abstract class NzTooltipBaseDirective implements OnChanges, OnDestroy, Af
...
+
+  ngOnInit(): void {
+    afterNextRender(
+      () => {
+        this.createComponent();
+        this.registerTriggers();
+      },
+      { injector: this.injector }
+    );
+  }
```
Comment (HyperLife1119) [inline components/tooltip/base.ts:133]: Key: afterNextRender will only be executed in the client, which ensures that components are only dynamically created in the browser.

File: components/tooltip/base.ts [hunk 4]
```diff
@@ -116,13 +114,21 @@ export abstract class NzTooltipBaseDirective implements OnChanges, OnDestroy, Af
...
+  protected platformId = inject(PLATFORM_ID);
+
+  constructor(protected componentType: Type<NzTooltipBaseComponent>) {}
+
+  ngAfterViewInit(): void {
+    if (isPlatformBrowser(this.platformId)) {
+      this.createComponent();
+      this.registerTriggers();
+    }
+  }
```
Comment (HyperLife1119) [inline components/tooltip/base.ts:131]: The reason for the hydration failure is that the host view node was removed using the native API when the component was dynamically created, resulting in the node not being found when hydrating.

While the issue can be resolved by using `ViewContainerRef.detach` instead of removeChild, this also results in components within the overlay not being able to correctly measure element sizes.

By controlling this part of the logic to be executed only in the client, we can temporarily solve this problem.

In conclusion, we should not remove the host view of a component after it has been rendered. The best solution here would be to use the `Overlay.create` API to render the component directly to the overlay container so that there is no need to `removeChild`.

**Review by Nicoss54 — APPROVED** [review]
LGTM

**Review by Laffery — APPROVED** [review]
LGTM too :)
